include/siminusminus/containers/inmutablestring.hpp

```diff
@@ -59,14 +64,15 @@ class InmutableString
      * Returns the i th position from our string.
      * @param index: index to string access.
      */
-    char operator[](const size_t index) const;
+    char& operator[](const size_t index) const;
```

Comment:
I have been reading this [ std::string operator[] ](http://es.cppreference.com/w/cpp/string/basic_string/operator_at) and returns a reference. I suppose it's wrong because you can modify the character and an InmutableString don't allow that, but if I return a const reference? I need it to do something like this: `InmutableIterator<char> iitbegin (&istring[0]);`